sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/encoders/ExpressionEncoder.scala

```diff
@@ -251,6 +251,27 @@ case class ExpressionEncoder[T](
 
     val plan = Project(Alias(unbound, "")() :: Nil, LocalRelation(schema))
     val analyzedPlan = SimpleAnalyzer.execute(plan)
+    if (!analyzedPlan.resolved) {
+      // We couldn't resolve, this means something is wrong with the schemas.
+      analyzedPlan match {
+        case p @ Project(projectList, _) => {
+          p.projectList.foreach { e: NamedExpression => e.foreach { _ match {
+            case u @ UpCast(child, dataType, walkedTypePath) => {
+              throw new AnalysisException(s"Cannot resolve `${child.prettyString}`. " +
+                "The type path of the target object is:\n" +
+                walkedTypePath.mkString("", "\n", "\n") +
+                "Ensure that the input schema contains this field.\n - Input schema is " +
+                schema.map(_.simpleString).mkString(", "))
```

Comment:
the `simpleString` of attribute includes expr id, which I think is noisy to users, should we generate the string manually or use `StructType.fromAttributes` to turn `schema` into `StructType` and call `StrcutType.simpleString`?